ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/Topic.java

```diff
@@ -58,11 +87,39 @@ public SerdeSupplier getSerdeSupplier() {
     return serdeSupplier;
   }
 
+  public SerdeFactory getKeySerdeFactory() {
+    return keySerdeFactory;
+  }
+
   Serializer getSerializer(final SchemaRegistryClient schemaRegistryClient) {
     return serdeSupplier.getSerializer(schemaRegistryClient);
   }
 
   Deserializer getDeserializer(final SchemaRegistryClient schemaRegistryClient) {
     return serdeSupplier.getDeserializer(schemaRegistryClient);
   }
+
+  Serializer getKeySerializer() {
+    return keySerdeFactory.create().serializer();
+  }
+
+  @SuppressWarnings("unchecked")
+  Deserializer getKeyDeserializer() {
+    final Serde keySerde = keySerdeFactory.create();
+    if (keySerde instanceof TimeWindowedSerde) {
+      final TimeWindowedSerde windowedSerde = (TimeWindowedSerde) keySerde;
+      final TimeWindowedDeserializer timeWindowedDeserializer =
+          (TimeWindowedDeserializer) windowedSerde.deserializer();
+      if (!windowSize.isPresent()) {
```

Comment:
nit: a few times in this method the if block is the bigger block. can we invert them?
```java
  Deserializer getKeyDeserializer() {
    final Serde keySerde = keySerdeFactory.create();
    if (!(keySerde instanceof TimeWindowedSerde)) {
      return keySerdeFactory.create().deserializer();
    }
    
    if (windowSize.isPresent()) {
      return new TimeWindowedDeserializer(new StringDeserializer(), windowSize.get());
    }

    final TimeWindowedSerde windowedSerde = (TimeWindowedSerde) keySerde;
    final TimeWindowedDeserializer timeWindowedDeserializer =
        (TimeWindowedDeserializer) windowedSerde.deserializer();
    if (timeWindowedDeserializer.getWindowSize() == Long.MAX_VALUE) {
      throw new KsqlException("Window size is not present for time windowed deserializer.");
    } 
    
    return new TimeWindowedDeserializer(
        new StringDeserializer(),
        timeWindowedDeserializer.getWindowSize());
  }
```

Also no need to call `.create()` multiple times if the first block (just call `keySerde.deserializer()`)